File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
    int n=nums.length;
    Map<Integer,Integer> map=new HashMap<>();
    int[] ans=new int[2];
    int i=0;
    while(i<n) {
        if (map.containsKey(target - nums[i])) {
            ans[1]=i;
            ans[0]=map.get(target - nums[i]);
            return ans;
        }
        map.put(nums[i], i);
        i++;
    }
    return ans;
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
    unordered_map<int, int> map;
    vector<int> ans(2);
    int n= nums.size();
    for(int i= 0;i<n;i++){
        if(map.find(target-nums[i])!=map.end()){
            ans[0]= map[target-nums[i]];
            ans[1]= i;
            return ans;
        }
        map[nums[i]]=i;
    }
    return ans;
    }    
};